sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -73,20 +73,29 @@ abstract class GpuParquetScanBase(
     readPartitionSchema: StructType,
     pushedFilters: Array[Filter],
     rapidsConf: RapidsConf,
-    supportsSmallFileOpt: Boolean)
+    supportsMultiFileOpt: Boolean,
+    canUseMultiThreadRead: Boolean,
+    canUseCoalesceFilesRead: Boolean)
   extends ScanWithMetrics with Logging {
 
+  def getCanUseCoalesceFilesRead: Boolean = canUseCoalesceFilesRead
+  def getSupportsMultiFileOpt: Boolean = supportsMultiFileOpt
```

Comment:
Rather than having these APIs, would it be easier and more flexible to expose the existing `rapidsConf` as a `val`?  If we decide to keep separate accessor methods, suggest ditching this and folding it into `getCanUseCoalesceFilesRead`

Reply:
same reason stated above, these are separate parameters due to the InputFileName issue where we find if after converting everything and then have to go back and switch it off. This is how it was done in the original change except now its the coalesce reader rather then all small file optimizations. I'll make sure to document it better.  Really this one we wouldn't need as we could just do the copy regardless of it is was on.